File: app/src/main/java/mega/privacy/android/app/fragments/settingsFragments/SettingsBaseFragment.java
package mega.privacy.android.app.fragments.settingsFragments;

import static mega.privacy.android.app.constants.EventConstants.EVENT_UPDATE_SCROLL;
import static mega.privacy.android.app.utils.Constants.SCROLLING_UP_DIRECTION;

import android.content.Context;
import android.os.Bundle;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.preference.Preference;
import androidx.preference.PreferenceFragmentCompat;
import androidx.recyclerview.widget.RecyclerView;

import com.jeremyliao.liveeventbus.LiveEventBus;

import mega.privacy.android.data.database.DatabaseHandler;
import mega.privacy.android.app.MegaApplication;
import mega.privacy.android.data.model.MegaPreferences;
import mega.privacy.android.app.interfaces.SimpleSnackbarCallBack;
import nz.mega.sdk.MegaApiAndroid;
import nz.mega.sdk.MegaChatApiAndroid;

public class SettingsBaseFragment extends PreferenceFragmentCompat implements Preference.OnPreferenceClickListener, Preference.OnPreferenceChangeListener {

    protected Context context;
    protected MegaApiAndroid megaApi;
    protected MegaChatApiAndroid megaChatApi;
    protected DatabaseHandler dbH;
    protected MegaPreferences prefs;

    protected SimpleSnackbarCallBack snackbarCallBack;

    public SettingsBaseFragment() {
        if (megaApi == null) {
            megaApi = MegaApplication.getInstance().getMegaApi();
        }

        if (megaChatApi == null) {
            megaChatApi = MegaApplication.getInstance().getMegaChatApi();
        }

        dbH = MegaApplication.getInstance().getDbH();
        prefs = dbH.getPreferences();
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        this.context = context;

        if (context instanceof SimpleSnackbarCallBack) {
            snackbarCallBack = (SimpleSnackbarCallBack) context;
        }
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        getListView().addOnScrollListener(new RecyclerView.OnScrollListener() {
            @Override
            public void onScrolled(@NonNull RecyclerView recyclerView, int dx, int dy) {
                super.onScrolled(recyclerView, dx, dy);
                LiveEventBus.get(EVENT_UPDATE_SCROLL, Boolean.class)
                        .post(recyclerView.canScrollVertically(SCROLLING_UP_DIRECTION));
            }
        });
    }

    @Override
    public void onCreatePreferences(Bundle bundle, String s) {
    }

    @Override
    public boolean onPreferenceClick(Preference preference) {
        return false;
    }

    @Override
    public boolean onPreferenceChange(Preference preference, Object newValue) {
        return false;
    }
}

File: app/src/main/java/mega/privacy/android/app/fragments/settingsFragments/SettingsBaseFragment.kt
package mega.privacy.android.app.fragments.settingsFragments

import android.content.Context
import android.os.Bundle
import android.view.View
import androidx.preference.PreferenceFragmentCompat
import androidx.recyclerview.widget.RecyclerView
import mega.privacy.android.app.activities.settingsActivities.PreferencesBaseActivity
import mega.privacy.android.app.interfaces.SimpleSnackbarCallBack
import mega.privacy.android.app.utils.Constants
import mega.privacy.android.data.database.DatabaseHandler
import mega.privacy.android.data.model.MegaPreferences
import mega.privacy.android.data.qualifier.MegaApi
import nz.mega.sdk.MegaApiAndroid
import nz.mega.sdk.MegaChatApiAndroid
import javax.inject.Inject

abstract class SettingsBaseFragment : PreferenceFragmentCompat() {

    @Inject
    @MegaApi
    internal lateinit var megaApi: MegaApiAndroid

    @Inject
    internal lateinit var megaChatApi: MegaChatApiAndroid

    @Inject
    internal lateinit var dbH: DatabaseHandler
    protected val prefs: MegaPreferences?
        get() = dbH.preferences
    protected var snackbarCallBack: SimpleSnackbarCallBack? = null

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is SimpleSnackbarCallBack) {
            snackbarCallBack = context
        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        listView.addOnScrollListener(object : RecyclerView.OnScrollListener() {
            override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
                super.onScrolled(recyclerView, dx, dy)
                (activity as? PreferencesBaseActivity)?.updateElevation(
                    recyclerView.canScrollVertically(
                        Constants.SCROLLING_UP_DIRECTION
                    )
                )
            }
        })
    }

}